Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -388,33 +388,61 @@ public class ProgramBuilder {
     /// Append a splice from another program.
     public func splice(from program: Program, at index: Int) {
         var idx = index
-        
+
+        // Get types of the other program
+        var ai = AbstractInterpreter(for: self.fuzzer)
+
+        for instr in program {
+            ai.execute(instr)
+        }
+
         // Determine all necessary input instructions for the choosen instruction
         // We need special handling for blocks:
         //   If the choosen instruction is a block instruction then copy the whole block
         //   If we need an inner output of a block instruction then only copy the block instructions, not the content
         //   Otherwise copy the whole block including its content
         var needs = Set<Int>()
         var requiredInputs = VariableSet()
-        
+
+        // This maps victim instruction indices to victim : host variable remap
+        // Instead of calling adopt and then using nextvar if the variable is
+        // not in the varMaps map, we do the adoption manually.
+        var rewireMapping: [Int: (Variable, Variable)] = [:]
+
+        func rewireOrKeepInputs(instruction: Instruction) {
+            var inputs: [Variable] = []
+            for input in instruction.inputs {
+                if probability(0.5) {
```

Comment:
My intuition is that this is too high, I'd think more of something in the range 0.10 to 0.25, but maybe this can be evaluated somehow? I'd guess a total replacement rate of maybe 10% would be reasonable

Reply:
Also I think it should only be done if `mode != .conservative`

Reply:
Yes, I think we should test different values and evaluate the effect. I think it would be good to be able to increase the aggressiveness over the course of a fuzzing session.

Reply:
I agree, we should only splice and re-wire if `mode == .aggressive`.

Reply:
Or like you said `if mode != .conservative` if we intend to add another even more aggressive mode.